### PR TITLE
py/emitnative.c: Fix exception handling issues triggered by uasyncio.

### PR DESCRIPTION
**Pull request by jimmo**
This fixes the issues that are currently causing the tests in #5332 to fail when run with the native emitter enabled.

Adds minimal tests to show the failures.

### CODE REVIEW

File: py/emitnative.c [hunk 1]
```diff
@@ -2058,7 +2058,7 @@ STATIC void emit_native_unwind_jump(emit_t *emit, mp_uint_t label, mp_uint_t exc
             ASM_MOV_REG_PCREL(emit->as, REG_RET, label & ~MP_EMIT_BREAK_FROM_FOR);
             ASM_MOV_LOCAL_REG(emit->as, LOCAL_IDX_EXC_HANDLER_UNWIND(emit), REG_RET);
             // Cancel any active exception (see also emit_native_pop_except_jump)
-            emit_native_mov_reg_const(emit, REG_RET, MP_F_CONST_NONE_OBJ);
+            ASM_MOV_REG_IMM(emit->as, REG_TEMP0, (mp_uint_t)MP_OBJ_NULL);
```
Comment (dpgeorge) [inline py/emitnative.c]: Should `REG_TEMP0` here be changed to `REG_RET`?

File: py/emitnative.c [hunk 2]
```diff
@@ -2154,7 +2154,7 @@ STATIC void emit_native_with_cleanup(emit_t *emit, mp_uint_t label) {
     ASM_MOV_REG_LOCAL(emit->as, REG_ARG_1, LOCAL_IDX_EXC_VAL(emit)); // get exc
 
     // Check if exc is None and jump to non-exc handler if it is
-    emit_native_mov_reg_const(emit, REG_ARG_2, MP_F_CONST_NONE_OBJ);
+    ASM_MOV_REG_IMM(emit->as, REG_ARG_2, (mp_uint_t)MP_OBJ_NULL);
     ASM_JUMP_IF_REG_EQ(emit->as, REG_ARG_1, REG_ARG_2, *emit->label_slot + 2);
```
Comment (dpgeorge) [inline py/emitnative.c]: I think these 2 lines can be optimised into 1 line using `ASM_JUMP_IF_REG_ZERO`.  We assume that `MP_OBJ_NULL` is zero.

Comment (jimmo) [reply py/emitnative.c]: Done. I set `bool_test` to `false` which I think is correct here (i.e. it's a strict non-zero test).

File: py/emitnative.c (hunk 1 above)
Comment (jimmo) [reply py/emitnative.c]: Yes. Fixed.

File: py/emitnative.c (hunk 2 above)
Comment (dpgeorge) [reply py/emitnative.c]: correct

**Comment by dpgeorge** [comment]
Really great, thank you!  Merged in 1f4b607116ea0f06111847510ba5b1cd8080bf7f through c3095b37e96aeb69564f53d30a12242ab42bbd02

**Pull request closed without merging**